pkg/operation/botanist/controlplane.go

```diff
@@ -1302,43 +1304,31 @@ func (b *Botanist) DeployKubeControllerManager(ctx context.Context) error {
 	return b.K8sSeedClient.ChartApplier().Apply(ctx, filepath.Join(chartPathControlPlane, v1beta1constants.DeploymentNameKubeControllerManager), b.Shoot.SeedNamespace, v1beta1constants.DeploymentNameKubeControllerManager, kubernetes.Values(values))
 }
 
-// DeployKubeScheduler deploys kube-scheduler deployment.
-func (b *Botanist) DeployKubeScheduler(ctx context.Context) error {
-	defaultValues := map[string]interface{}{
-		"replicas":          b.Shoot.GetReplicas(1),
-		"kubernetesVersion": b.Shoot.Info.Spec.Kubernetes.Version,
-		"podAnnotations": map[string]interface{}{
-			"checksum/secret-kube-scheduler":        b.CheckSums[v1beta1constants.DeploymentNameKubeScheduler],
-			"checksum/secret-kube-scheduler-server": b.CheckSums[common.KubeSchedulerServerName],
-		},
-		"podLabels": map[string]interface{}{
-			v1beta1constants.LabelPodMaintenanceRestart: "true",
-		},
-	}
-
-	if b.ShootedSeed != nil {
-		defaultValues["resources"] = map[string]interface{}{
-			"limits": map[string]interface{}{
-				"cpu":    "300m",
-				"memory": "512Mi",
-			},
-		}
+// DefaultKubeScheduler returns a deployer for the kube-scheduler.
+func (b *Botanist) DefaultKubeScheduler() (kubescheduler.KubeScheduler, error) {
+	image, err := b.ImageVector.FindImage(common.KubeSchedulerImageName, imagevector.RuntimeVersion(b.SeedVersion()), imagevector.TargetVersion(b.ShootVersion()))
+	if err != nil {
+		return nil, err
 	}
 
-	if schedulerConfig := b.Shoot.Info.Spec.Kubernetes.KubeScheduler; schedulerConfig != nil {
-		defaultValues["featureGates"] = schedulerConfig.FeatureGates
-
-		if schedulerConfig.KubeMaxPDVols != nil {
-			defaultValues["kubeMaxPDVols"] = *schedulerConfig.KubeMaxPDVols
-		}
-	}
+	return kubescheduler.New(
+		b.K8sSeedClient.DirectClient(),
+		b.Shoot.SeedNamespace,
+		b.Shoot.Info.Spec.Kubernetes.Version,
+		image.String(),
+		int32(b.Shoot.GetReplicas(1)),
```

Comment:
Can you change the return type of `GetReplicas` to `int32`?
```suggestion
		b.Shoot.GetReplicas(1),
```